File: euler/problem1/python/e.py
import time

def gauss_sum(val):
    p = limit//val
    return val*p*(p+1)//2

# Record the start time
acc = 0
limit = 999
start_time = time.perf_counter()

for i in range(1,limit+1):
    if (i%3==0) or (i%5==0):
        acc+=i
end_time = time.perf_counter()

print(str(acc) + "\nfor loop\nElapsed time: " + str(end_time - start_time))

start_time = time.perf_counter()
res = gauss_sum(3) + gauss_sum(5) - gauss_sum(15)

end_time = time.perf_counter()

print(str(res) + "\nGauss formula\nElapsed time: " + str(end_time - start_time))

File: euler/problem1/c++/e.cpp
#include <iostream>
#include <chrono>

int limit = 999;

int gauss_sum(int x) {
	int p = limit/x;
	return x*(p*(p+1))/2;
}

int main() {
	int result = 0;
	
	auto start = std::chrono::high_resolution_clock::now();
	for (int i = 0; i < limit+1; i++)
		if ((i%3==0) || (i%5==0))
			result = result + i;
	
	auto end = std::chrono::high_resolution_clock::now();
	auto duration = std::chrono::duration_cast<std::chrono::nanoseconds>(end - start);

	std::cout << "Result: " << result << "\nfor cicle"<< std::endl;
	std::cout << "Elapsed time: " << duration.count() << " nanoseconds" << std::endl;


	start = std::chrono::high_resolution_clock::now();
	
	result = gauss_sum(3) + gauss_sum(5) - gauss_sum(15);

	end = std::chrono::high_resolution_clock::now();
	duration = std::chrono::duration_cast<std::chrono::nanoseconds>(end - start);


	std::cout << "Result: " << result << "\nGauss formula" << std::endl;
    	std::cout << "Elapsed time: " << duration.count() << " nanoseconds" << std::endl;

	return 0;
}
